src/conda_package_handling/conda_fmt.py

```diff
@@ -100,9 +115,20 @@ def create(
             # put the info last, for parity with updated transmute.
             compress = compressor()
             for component, files in components_files:
+
+                # If size is known, the decompressor may be able to allocate less memory.
+                # The compressor will error if size is not correct.
+                with tarfile.TarFile(fileobj=NullWriter(), mode="w") as sizer:  # type: ignore
+                    for file in files:
+                        sizer.add(file, filter=utils.anonymize_tarinfo)
+                    sizer.close()
```

Comment:
Do we need an extra `close()` here if the with statement also closes `sizer`?

Reply:
Oh, Python really doesn't like creating scopes. So we can just access sizer below the with statement.